lib/mutant/repository/diff.rb

```diff
@@ -31,15 +31,21 @@ def touches?(path, line_range)
 
       # rubocop:disable Metrics/MethodLength
       def touched_paths
-        pathname = world.pathname
-        work_dir = pathname.pwd
+        root_stdout =
+          world
+            .capture_stdout(%w[git rev-parse --show-toplevel])
+            .from_right
+            .lines
```

Comment:
The `lines.first` combo is also IMO not a good idea. We know t hat `git rev-parse --show-toplevel` is supposed to print a single line. If it prints more than one line we actually "want to blow up" and not silently truncate all but the first lines.

Also `.first` in case no line is printed: would result in `nil` which is hard to debug.

Short of parsing the entire output, I think we should do the minimal transformation (truncating the `"\n"`) and bubble up the result. This will make the "wrong" values noticable at least.

OT: my haskell mutant main process does this correctly, its still not release ready unfortunately.

Reply:
good point on `nil`